grpc/src/main/java/com/linecorp/armeria/server/grpc/DefaultUnframedGrpcErrorHandler.java

```diff
@@ -0,0 +1,289 @@
+                 JsonGenerator jsonGenerator = mapper.createGenerator(outputStream)) {
+                jsonGenerator.writeStartObject();
+                jsonGenerator.writeFieldName("error");
+                jsonGenerator.writeStartObject();
+                jsonGenerator.writeNumberField("code", grpcCode.value());
+                if (grpcMessage != null) {
+                    jsonGenerator.writeStringField("message", grpcMessage);
+                }
+                if (cause != null && ctx.config().verboseResponses()) {
+                    jsonGenerator.writeStringField("stack-trace", Exceptions.traceText(cause));
+                }
+                if (!Strings.isNullOrEmpty(grpcStatusDetailsBin)) {
+                    com.google.rpc.Status rpcStatus = null;
+                    try {
+                        rpcStatus = decodeGrpcStatusDetailsBin(grpcStatusDetailsBin);
+                    } catch (InvalidProtocolBufferException e) {
+                        logger.warn("Unexpected exception while decoding grpc-status-details-bin: {}",
+                                    grpcStatusDetailsBin, e);
+                    }
+                    if (rpcStatus != null) {
+                        jsonGenerator.writeFieldName("details");
+                        writeErrorDetails(rpcStatus.getDetailsList(), jsonGenerator);
+                    }
+                }
+                jsonGenerator.writeEndObject();
+                jsonGenerator.writeEndObject();
+                jsonGenerator.flush();
+                success = true;
+            } catch (IOException e) {
+                logger.warn("Unexpected exception while generating a JSON response", e);
+            } finally {
+                if (!success) {
+                    buffer.release();
+                }
+            }
+            if (success) {
+                return HttpResponse.of(responseHeaders, HttpData.wrap(buffer));
+            } else {
+                return HttpResponse.of(responseHeaders);
+            }
+        };
+    }
+
+    private DefaultUnframedGrpcErrorHandler() {}
+
+    /**
+     * Ensure that unframedGrpcStatusMappingFunction never returns null by falling back to the default.
+     */
+    private static UnframedGrpcStatusMappingFunction ofStatusMappingFunction(
+            UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        requireNonNull(statusMappingFunction, "statusMappingFunction");
+        if (statusMappingFunction == UnframedGrpcStatusMappingFunction.of()) {
+            return statusMappingFunction;
+        }
+        return statusMappingFunction.orElse(UnframedGrpcStatusMappingFunction.of());
+    }
+
+    @VisibleForTesting
+    static void writeErrorDetails(List<Any> details, JsonGenerator jsonGenerator) throws IOException {
+        jsonGenerator.writeStartArray();
+        for (Any detail : details) {
+            try {
+                ERROR_DETAILS_MARSHALLER.writeValue(detail, jsonGenerator);
```

Comment:
Users might want to register a cusom `Message` to support the serialization. Let's leave it as a known issue and implement it later if there's a demand for that. 👍